service/src/test/java/bio/terra/drshub/services/MetadataServiceTest.java

```diff
@@ -0,0 +1,59 @@
+package bio.terra.drshub.services;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+import bio.terra.drshub.BaseTest;
+import bio.terra.drshub.models.DrsApi;
+import io.github.ga4gh.drs.model.Authorizations;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.web.client.RestClientException;
+
+public class MetadataServiceTest extends BaseTest {
+
+  @Autowired private MetadataService metadataService;
+  @MockBean private DrsApi drsApi;
+
+  @MockBean private DrsApiFactory drsApiFactory;
+
+  @Test
+  void testDrsOptionsEndpoint() {
```

Comment:
Looks like this is testing 3 different cases. I think I've typically seen our unit tests organized so that they test specific behavior. So instead of testing 3 cases in one test, this would have been 3 unit tests to test out each "unit" of behavior. Any thoughts on that / why you decided to put all 3 in one test?

Reply:
There are different ways of organizing unit tests. I like to organize tests by the method they're testing, since a lot of the time, there's mocking or setup that needs to be done. Replicating that setup over many tests make refactoring harder, especially if you need to update the tests for a method!